beeline/__init__.py

```diff
@@ -164,22 +176,42 @@ def _run_hooks_and_send(self, ev):
         ''' internal - run any defined hooks on the event and send '''
         presampled = False
         if self.sampler_hook:
+            self.log("executing sampler hook on event ev = %s", ev.fields())
             keep, new_rate = self.sampler_hook(ev.fields())
             if not keep:
+                self.log("skipping event due to sampler hook sampling ev = %s", ev.fields())
                 return
             ev.sample_rate = new_rate
             presampled = True
 
         if self.presend_hook:
+            self.log("executing presend hook on event ev = %s", ev.fields())
             self.presend_hook(ev.fields())
 
         if hasattr(ev, 'traced_event'):
+            self.log("enqueing traced event ev = %s", ev.fields())
             self.tracer_impl.send_traced_event(ev, presampled=presampled)
         elif presampled:
+            self.log("enqueing presampled event ev = %s", ev.fields())
             ev.send_presampled()
         else:
+            self.log("enqueing event ev = %s", ev.fields())
             ev.send()
 
+    def _init_logger(self):
+        import logging
+        self._logger = logging.getLogger('honeycomb-beeline')
+        self._logger.setLevel(logging.DEBUG)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.DEBUG)
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        ch.setFormatter(formatter)
+        self._logger.addHandler(ch)
+
+    def log(self, msg, *args, **kwargs):
+        if self.verbose_mode:
```

Comment:
Do you need this conditional since if verbose mode is not set logger will not be initialized? 
Maybe initialize a default null logger and then _init_logger can replace it with a real logger? 
It feels odd to have the conditional both in initializing the logger and using the logger.

Reply:
Hmm.  Maybe I should rescind this comment. There's value (especially in the SDK) to not initializing a logger if you're not going to use one. And there's value to consistency between the two. The double check still feels a bit odd but I defer to other eng for whether it's actually odd or just me.